src/Nethermind/Nethermind.Merge.Plugin/Data/ExecutionPayload.cs

```diff
@@ -195,11 +175,17 @@ public static class ExecutionPayloadExtensions
         int version,
         [NotNullWhen(false)] out string? error)
     {
+        if (spec.IsEip4844Enabled && executionPayload is not ExecutionPayloadV3)
```

Comment:
If we are going class hierarchy route, maybe `Validate` should be a virtual method on `ExcecutionPayload`?